apps/www/_blog/2023-12-01-automatic-cli-login.mdx

```diff
@@ -0,0 +1,55 @@
+---
+
+Using the [Supabase CLI](https://supabase.com/docs/guides/cli) is the go-to method for many to interact with their Supabase projects. It enables easy local project development, working with the local database, managing migrations, and generating types, among other functionalities. However, some scenarios demand connecting to the real Supabase project, requiring a login. Until now, the login process involved typing `supabase login`, which prompted users to manually generate a token from https://supabase.com/dashboard/account/tokens and paste it back into the terminal.
+
+This method proved inconvenient and suboptimal for our users, especially for those recording courses or tutorial videos, as they repeatedly had to go through the whole process manually. We are committed to prioritizing Developer Experience (DX), so we decided to improve this.
+
+But what if logging in were as simple as typing `supabase login` and, if you were already logged into your dashboard, everything happened automatically? This streamlined approach began with `1.114.0`.
+
+## How does Automatic CLI login work exactly?
+
+When you type `supabase login`, several things occur. First, a random UUID is generated to represent your unique login attempt. Next, a token name is created using your username, machine name, and the current timestamp. Most importantly, a pair of private/public P-256 elliptic curve Diffie-Hellman (ECDH) keys is generated. These keys remain valid only for the duration of the process; once you log in or cancel the CLI process, they become unusable.
+
+<div className="video-container">
+  <iframe
+    className="w-full"
+    src="https://www.youtube-nocookie.com/embed/lNgsIITU3pc?si=X6KE7X819Hk"
+    title="YouTube video player"
+    frameBorder="0"
+    allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
+    allowfullscreen
+  ></iframe>
+</div>
+
+After gathering all this data, a unique URL is generated containing the session ID, token name, and your public key. The CLI then attempts to open this URL in your current environment's default browser. Once it opens, it verifies your authentication. If you're not authenticated, you're redirected to the login page, prompted to log into the dashboard, and upon successful authentication, redirected back to the originally requested page.
+
+With your identity confirmed, the CLI sends the provided session data to our management API, requesting its magic.
+
+The Management API generates a new access token for your account, creates its pair of ECDH keys, and encrypts the token using your public key and the server's private key. This encrypted token, alongside the server's public key, your session ID, and expiration date, is then stored in our database.
+
+The Management API generates a new access token for your account. To send it back to your CLI it needs to be stored in a database.
+
+## Token encryption with Elliptic-Curve Diffie-Hellman
```

Comment:
Maybe:

```suggestion
## Securely exchanging the token using ECDH
```

Technically the encryption is not ECDH -- ECDH is just the method to agree on a shared encryption key. I have a feeling HN will point this out. 😂